src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+          $this->addFileDataToJob($remote->getJob(), $status, $file_id);
+          $translated++;
+        }
+        else {
+          $not_translated++;
+        }
+      }
+    }
+    catch (TMGMTException $e) {
+      drupal_set_message(new TranslatableMarkup('Could not pull translation resources due to the following error: @message', ['@message' => $e->getMessage()]), 'error');
+    }
+    finally {
+      return [
+        'updated' => $translated,
+        'non-updated' => $not_translated,
+      ];
+    }
+  }
+
+  /**
+   * Returns the file info of one state.
+   *
+   * @param string $state
+   *   The state.
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   (Optional) A Job.
+   *
+   * @return array
+   *   The file infos.
+   */
+  protected function requestFileinfo($state, JobInterface $job = NULL) {
+    $all_files = [];
+    $files = $this->request('fileinfos/' . $state);
+    foreach ($files as $file) {
+      $all_files[$file['FileId']] = $file;
+    }
+    if ($job) {
+      $files = [];
+      $mappings = RemoteMapping::loadByLocalData($job->id(), NULL, 'tmgmt_thebigword');
+      /** @var RemoteMapping $mapping */
+      foreach ($mappings as $mapping) {
+        $remote_files = $mapping->getRemoteData('files');
+        foreach ($remote_files as $file_id => $file) {
+          if (isset($all_files[$file_id]) && $all_files[$file_id]['FileStateVersion'] == $file['FileStateVersion']) {
+            $files[$file_id] = $all_files[$file_id];
+          }
+        }
+      }
+    }
+    else {
+      $files = $all_files;
+    }
+    return $files;
+  }
+
+  /**
+   * Sends an error file to Thebigword.
+   *
+   * @param string $state
+   *   The state.
+   * @param string $file_id
+   *   The file id to update.
+   * @param string $message
+   *   The error message.
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   (Optional) A Job, if a JobItem is given, Job will be discarded.
+   *   At least or a Job or a JobItem must be given.
+   * @param \Drupal\tmgmt\JobItemInterface $job_item
+   *   (Optional) A JobItem.
+   *   At least or a Job or a JobItem must be given.
+   * @param bool $confirm
+   *   (Optional) Set to TRUE if also want to send the confirmation message
+   *   of this error. Otherwise will not send it.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   *   If there is a problem with the request or if no Job and no JobItem are
+   *   given.
+   */
+  public function sendFileError($state, $file_id, $message = '', JobInterface $job = NULL, JobItemInterface $job_item = NULL, $confirm = FALSE) {
+    if (!$job_item && !$job) {
+      throw new TMGMTException('No Job or JobItem given to sendFileError, at least one of both arguments is mandatory.');
+    }
+    if ($job_item) {
+      $job = $job_item->getJob();
+    }
+    $remote_mappings = RemoteMapping::loadByLocalData($job->id(), $job_item != NULL ? $job_item->id() : NULL, 'tmgmt_thebigword');
```

Comment:
same here, if the only thing that is required is the project_id, then have that as a requirement is the project_id, then you can delete a lot of code here?